Mathlib/Algebra/Regular/SMul.lean

```diff
@@ -231,13 +229,22 @@ section SMulZeroClass
 
 variable {M}
 
-protected
-lemma IsSMulRegular.eq_zero_of_smul_eq_zero [Zero M] [SMulZeroClass R M]
+protected lemma IsSMulRegular.eq_zero_of_smul_eq_zero [Zero M] [SMulZeroClass R M]
     {r : R} {x : M} (h1 : IsSMulRegular M r) (h2 : r • x = 0) : x = 0 :=
   h1 (h2.trans (smul_zero r).symm)
 
 end SMulZeroClass
 
+lemma isSMulRegular_iff_smul_eq_zero_imp_eq_zero [AddGroup M] [DistribSMul R M] (r : R) :
+    IsSMulRegular M r ↔ ∀ m : M, r • m = 0 → m = 0 where
```

Comment:
```suggestion
lemma isSMulRegular_iff_right_eq_zero_of_smul [AddGroup M] [DistribSMul R M] (r : R) :
    IsSMulRegular M r ↔ ∀ m : M, r • m = 0 → m = 0 where
```

Reply:
[isSMulRegular_iff_smul_eq_zero_imp_eq_zero](https://leanprover-community.github.io/mathlib4_docs/Mathlib/RingTheory/Regular/IsSMulRegular.html#isSMulRegular_iff_smul_eq_zero_imp_eq_zero) is pre-existing. Should I deprecate it?